greedy_algorithms/best_time_to_buy_and_sell_stocks_1.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+Best Time to Buy and Sell Stock - Greedy
+
+Problem Statement: 
+Given an array for which the ith element is the price of a given stock on day i.
+If you were only permitted to complete at most one transaction 
+(i.e., buy one and sell one share of the stock), design an algorithm to find the maximum profit.
+Note that you cannot sell a stock before you buy one.
+
+Solution:
+
+we find the maximum price for which the stock can be sell out at ith day in vector b, ith element of b
+is the maximum stock price from (n-1)th day to ith day. Then for each day find the profit by subtracing 
+the cost of buying at ith day, final answer will be the maximum of these profits.
+*/
+
+
+#include <iostream>
+#include <vector> 
+#include <cassert>
+
+int maxProfit(std::vector<int> prices, int n) 
```

Comment:
- Please document the function signature.
- `n` seems redundant, `prices.size()` should be preferable.
- Pass the input vector as const reference to avoid copying.